### ISSUE TITLE
math not processing correctly

### ISSUE THREAD

**Issue by MichaelJCompton** [dgraph-io/dgraph#942]
Running https://docs.dgraph.io/v0.7.6/query-language/#aggregating-value-variables example 2 on master fails.  The `score` doesn't calculate.

An extra variable (here `data`) accumulating the result makes it work, but it doesn't work in the one math expression.
```
{
  q(func:allofterms(name, "steven spielberg")) {
    name@en
    films as director.film {
      p as count(starring)
      q as count(genre)
      
      date as initial_release_date
      years as math(since(date)/(365*24*60*60))

      data as math(ln(p)+q-ln(years))
      score as math(cond(years > 10, 0, data))
    }
  }

  TopMovies(id: var(films), orderdesc: var(score), first: 5){
    name@en
    var(score)
    var(date)
  }
}
```

**Comment by pawanrawal** [comment]
@MichaelJCompton 

```
{
  q(func:allofterms(name, "steven spielberg")) {
    name@en
    films as director.film {
      p as count(starring)
      q as count(genre)
      
      date as initial_release_date
      years as math(since(date)/(365*24*60*60))

      score as math(cond(years > 10, 0, ln(p)+q-ln(years)))
    }
  }

  TopMovies(id: var(films), orderdesc: var(score), first: 5){
    name@en
    var(score)
    var(date)
  }
}
```
seems to work. Could you please share the query that doesn't work?

**Comment by MichaelJCompton** [comment]
Hi @pawanrawal ,

I'm running

Dgraph version   : v0.7.6-dev
Commit SHA-1     : 913f40d
Commit timestamp : 2017-05-12 17:03:34 +1000
 
the query you posted doesn't work for me - the query runs but it doesn't calculate `score`.

With the extra variable it does though.

**Comment by ashwin95r** [comment]
Fixed and merged to release/v0.7.7.
